torchao/utils.py

```diff
@@ -22,19 +24,49 @@
 
 
 def benchmark_model(model, num_runs, input_tensor):
-    torch.cuda.synchronize()
-    start_event = torch.cuda.Event(enable_timing=True)
-    end_event = torch.cuda.Event(enable_timing=True)
-    start_event.record()
-
-    # benchmark
-    for _ in range(num_runs):
-        with torch.autograd.profiler.record_function("timed region"):
-            model(input_tensor)
-
-    end_event.record()
-    torch.cuda.synchronize()
-    return start_event.elapsed_time(end_event) / num_runs
+    if _assert_and_get_unique_device(model).type == "cuda":
+        torch.cuda.synchronize()
+        start_event = torch.cuda.Event(enable_timing=True)
```

Comment:
@albanD is there a reason why we have device specific events in PyTorch, it feels like this code would be considerably simpler if we just had a notion of an Event that prioritized cuda devices then mps then CPU

Reply:
If you're running latest PyTorch, do use `torch.Event` !